### ISSUE TITLE
Support for ZRANGESTORE

### ISSUE THREAD

**Issue by danielgerlag** [StackExchange/StackExchange.Redis#2047]
Are there any plans to implement support for ZRANGESTORE?

https://redis.io/commands/zrangestore

**Comment by NickCraver** [comment]
No immediate plans at the moment but seems on the simpler side since it's a new command (e.g. unlike `ZRANGE`/`ZREVRANGE` it's an `int` return) - happy to take a PR or I'll try and get to this in the queue.

One thing that concerns me is that the redis.io prompt doesn't have it working which I haven't seen before /cc @slorello89: 
<img width="336" alt="image" src="https://user-images.githubusercontent.com/454813/159126184-a1b5e458-f134-4369-aedc-d78821167ebd.png">

**Comment by danielgerlag** [comment]
I think that prompt is backed by v6.0. This command is new in v6.2.

**Comment by NickCraver** [comment]
Gotcha, thanks. Given how many commands are added in v6.2.0, I want to see if we can change that - as a library author it really helps in playing with things quickly.

**Comment by slorello89** [comment]
Hey @NickCraver , it looks like it's not an issue with the version or the command or anything, just a bit of weirdness in redis-io. 

TL;DR: There's a missing line in a config file in the reids-io app that's causing that weirdness

Basically, the redis-io interactive uses a bit of tomfoolery to let you think it's a straight pass-through to redis, however it intercepts your commands and uses your sessionId to namespace all the keys you're interacting with from it. The reason for namespacing everything is so that when your session dies it can tare out all the keys you added to Redis, as you can imagine it's a security/stability concern to leave an open-ended terminal to Redis :). There's a [config file](https://github.com/redis/redis-io/blob/master/lib/interactive/commands.rb) (well it's code but let's call it a config file) that tells redis-io how to parse and namespace the arguments for each command. If the command isn't in that file it just assumes the command either doesn't exist or is explicitly in the don't-allow list. You'll notice that `ZRANGESTORE` isn't in there. I'll check with @itamarhaber why this is, it should be auto-generated by a [script](https://github.com/redis/redis-io/blob/master/scripts/generate_interactive_commands.rb), but I think a confluence of events prevented it from making it in. Looks like the `ZRANGESTORE` command missed the train by a few weeks in redis/redis-io#226, which added all the 6.2 - looks like zrangestore was added a few weeks later in redis/redis/#7844 - also looks like there might be a bit of an issue with the script which should autogenerate everything (could just be my environment I'm very much not a rubiest but there are parsing issues from the commands.json file. 

Regardless, if you add `"zrangestore" => [:custom],` to `commands.rb` and spin up redis-io it works as you'd expect:

![image](https://user-images.githubusercontent.com/42971704/159282950-c734a152-10bf-423d-a049-00021d06c2a4.png)

So I wouldn't let this weirdness in redis-io dissuade you from including `ZRANGESTORE` in the library.

I'll track down how to fix the interactive :) 

in the interim @danielgerlag - you can of course use the ad-hoc commands `db.Execute`/`db.ExecuteAsync`. In my experience they tend to work just fine, you don't have a particularly complicated result to parse as it just returns an int.

```csharp
await db.SortedSetAddAsync("srczset", new SortedSetEntry[]
{
    new("one",1),
    new("two", 2),
    new ("three", 3),
    new("four", 4)
});

var numInSet = await db.ExecuteAsync("ZRANGESTORE", "dstzset", "srczset", 2, -1);

Console.WriteLine((int)numInSet);

var range = await db.SortedSetRangeByRankAsync("dstzset");

foreach (var elem in range)
{
    Console.WriteLine((string)elem);
}
```

does the job.

I'd be happy to open a PR for this if you'd like @NickCraver?

**Comment by NickCraver** [comment]
Ahh awesome, neat to see how that works - thanks @slorello89! Yep, a PR is certainly welcome, or I'll get to it this week or next. I want to tackle the cluster error detection soon (e.g. primary/replica swap for a member) and I've got a few other PRs in flight, just going down the queue :)

**Comment by slorello89** [comment]
Btw @NickCraver - the console on redis.io should be fixed now.
